src/distributions/mod.rs

```diff
@@ -77,7 +77,7 @@ impl<Sup> RandSample<Sup> {
 /// A value with a particular weight for use with `WeightedChoice`.
 pub struct Weighted<T> {
     /// The numerical weight of this item
-    pub weight: usize,
+    pub weight: u64,
```

Comment:
Hm ok, for now let's stick to `u32` for this as I think it's more in accordance with libstd's guidelines, but I think after that we're good to go